### PR TITLE
glowup: prefect-server chart

### LINKED ISSUE #107: Admin DB user vs Orion DB user distinction / documentation
I have found the comments in the values.yml regarding database users a little confusing. I would have expected that you launch the chart providing, potentially, admin credentials to a database server, and the chart creates a database, a user, and a password to access this database.

However, when looking at the comments https://github.com/PrefectHQ/prefect-helm/blob/main/charts/prefect-orion/values.yaml#L219, https://github.com/PrefectHQ/prefect-helm/blob/main/charts/prefect-orion/values.yaml#L227, I am not fully clear whether these values are for the admin user or for the "application" user

Thank you for your help

### LINKED ISSUE #187: Improve default server chart installation experience
**Summary**

The default Helm chart install experience for Prefect server requires that you specify a database password (or disable PostgreSQL) and configure the publicApiUrl.

Since our default experience recommends using port forwarding, we should tailor the defaults to "just work" out of the box for that situation.

**Current behavior**

Attempting to install the chart without any parameters results in an error. In general, if possible, we should try to install something that works (with reasonable defaults), to streamline the experience of getting started.

```
$ helm install server-test ./charts/prefect-server
Error: INSTALLATION FAILED: execution error at (prefect-server/templates/secret.yaml:18:24): .Values.postgresql.auth.password is required.
```

Additionally, the instructions in the notes suggest a series of commands to get the pod name, container port, and configure port forwarding:

```
1. Get the application URL by running these commands:
  export POD_NAME=$(kubectl get pods --namespace jawnsy-test -l "app.kubernetes.io/name=prefect-server,app.kubernetes.io/instance=server-test" -o jsonpath="{.items[0].metadata.name}")
  export CONTAINER_PORT=$(kubectl get pod --namespace jawnsy-test $POD_NAME -o jsonpath="{.spec.containers[0].ports[0].containerPort}")
  echo "Visit http://127.0.0.1:8080 to use your application"
  kubectl --namespace jawnsy-test port-forward $POD_NAME 8080:$CONTAINER_PORT
```

This is much more complicated than it needs to be.

**Desired behavior**

The goal is to have `helm install ./charts/prefect-server` do something meaningful with default values.

For the PostgreSQL password issue, we can either:

1. Default to SQLite instead of PostgreSQL (that is, change the default `postgresql.enabled` to `false`); or,
2. Generate a PostgreSQL password if none is specified

For the `port-forward` issue, we can default to `http://localhost:4200/api` as our default `server.publicApiUrl` and we can use the `kubectl` shortcut based on our deployment name:

```
kubectl --namespace jawnsy-test port-forward deployment/prefect-server 4200:4200
```

### LINKED ISSUE #199: Issues Connecting Prefect to Postgresql
I'm trying to connect with Prefect, I passing a URL with the following format to the secret documented [here](https://github.com/PrefectHQ/prefect-helm/blob/main/charts/prefect-server/values.yaml#L268):

```
postgresql+asyncpg://user:pass@mydb-host.us-west-2.rds.amazonaws.com/dbname?sslmode=require&sslrootcert=~/.postgresql/server-ca.pem 
```

I'm mounting the certificates in a `volumeMount` in the path `~/.postgresql` as documented [here](https://www.postgresql.org/docs/current/libpq-ssl.html).

This configuration works fine without the SSL configuration but it seems like there is a lack of documentation when it comes to using SSL to connect to postgres.

When I use the URL above, The prefect-server pod fails with the following error:

```
  File "/usr/local/lib/python3.8/site-packages/sqlalchemy/engine/default.py", line 615, in connect
    return self.loaded_dbapi.connect(*cargs, **cparams)
  File "/usr/local/lib/python3.8/site-packages/sqlalchemy/dialects/postgresql/asyncpg.py", line 916, in connect
    await_only(creator_fn(*arg, **kw)),
TypeError: connect() got an unexpected keyword argument 'sslmode'

Application startup failed. Exiting.
```

According to [this issue](https://github.com/sqlalchemy/sqlalchemy/issues/6275), the key `ssl` should be used instead of  `sslmode` but still failing this time with the error:

```
    return self.loaded_dbapi.connect(*cargs, **cparams)
  File "/usr/local/lib/python3.8/site-packages/sqlalchemy/dialects/postgresql/asyncpg.py", line 916, in connect
    await_only(creator_fn(*arg, **kw)),
TypeError: connect() got an unexpected keyword argument 'sslrootcert'
```

How can I configure Prefect to use an SSL connection with Postgresql using certificates? 

Thanks!

This issue can be related to https://github.com/sqlalchemy/sqlalchemy/issues/6275

### PR DESCRIPTION
**Pull request by jamiezieziula**
Resolves https://github.com/PrefectHQ/prefect-helm/issues/187
Resolves https://github.com/PrefectHQ/prefect-helm/issues/107
Resolves https://github.com/PrefectHQ/prefect-helm/issues/199 - Thanks to @MrCoffey for the research & documentation contribution!

## Default postgresql password & API URL
```shell
jamiedick in ~/src/prefecthq/prefect-helm/charts/prefect-server on server-work λ helm install server .
NAME: server
LAST DEPLOYED: Wed Jul 26 22:15:07 2023
NAMESPACE: prefect
STATUS: deployed
REVISION: 2
TEST SUITE: None
NOTES:
Run the following command to port-forward the UI to your localhost:
$ kubectl --namespace jamie-helm-test port-forward svc/prefect-server 4200:4200

Visit http://localhost:4200 to use Prefect!
jamiedick in ~/src/prefecthq/prefect-helm/charts/prefect-server on server-work λ kubectl port-forward svc/prefect-server 4200:4200
Forwarding from 127.0.0.1:4200 -> 4200
Forwarding from [::1]:4200 -> 4200
```
Screenshot 2023-07-26 at 10 14 45 PM

Also tested with an existing secret and everything looks good

### CODE REVIEW

File: charts/prefect-server/values.yaml [hunk 1]
```diff
@@ -36,8 +36,8 @@ server:
   # -- priority class name to use for the server pods; if the priority class is empty or doesn't exist, the server pods are scheduled without a priority class
   priorityClassName: ""
 
-  # -- sets PREFECT_UI_API_URL; should be publicly accessible API URL; UI will not work unless set
-  publicApiUrl: ""
+  # -- sets PREFECT_UI_API_URL; should be publicly accessible API URL
+  publicApiUrl: http://localhost:4200/api
```
Comment (jamiezieziula) [inline charts/prefect-server/values.yaml:40]: providing a default value to for the UI url for ease of new users

File: charts/prefect-server/values.yaml [hunk 2]
```diff
@@ -169,7 +169,7 @@ service:
   port: 4200
   # -- service Cluster IP
   clusterIP: ""
-  # -- service port if defining service as type nodeport
+  # -- service port if defining service as type nodeport TEST
```
Comment (jamiezieziula) [inline charts/prefect-server/values.yaml:172]:
```suggestion
  # -- service port if defining service as type nodeport
```

File: charts/prefect-server/values.yaml [hunk 3]
```diff
@@ -247,22 +247,21 @@ ingress:
...
-    # -- name for a custom user to create
+    # -- name for a custom user
     username: prefect
     ## This is the password for `username` and will be set within the secret `{fullnameOverride}-postgresql` at the key `password`.
     ## This argument is only relevant when using the Postgres database included in the chart.
     ## For an external postgres connection, you must create and use `existingSecret` instead.
-    # -- password for the custom user to create. Ignored if `auth.existingSecret` with key `password` is provided
-    password: ""
+    # -- password for the custom user. Ignored if `auth.existingSecret` with key `password` is provided
+    password: "prefect-rocks"
```
Comment (jamiezieziula) [inline charts/prefect-server/values.yaml:264]: providing a default password so that chart install doesnt fail if the user doesn't make any changes

**Review by parkedwards — DISMISSED** [review]
this is awesome

**Review by parkedwards — APPROVED** [review]

**Review by jimid27 — APPROVED** [review]
this is spectacular

**Pull request merged**